Example/Example/StringUtilities.swift

```diff
@@ -0,0 +1,30 @@
+//
+//  StringUtilities.swift
+//  Example
+//
+//  Created by Mikhail on 05/10/16.
+//  Copyright © 2016 Polidea. All rights reserved.
+//
+
+import Foundation
+extension String {
+    
+    /// Create `NSData` from hexadecimal string representation
+    ///
+    /// This takes a hexadecimal representation and creates a `NSData` object. Note, if the string has any spaces or non-hex characters (e.g. starts with '<' and with a '>'), those are ignored and only hex characters are processed.
+    ///
+    /// - returns: Data represented by this hexadecimal string.
+    
+    func dataFromHexadecimalString() -> NSData? {
+        let data = NSMutableData(capacity: characters.count / 2)
+        
+        let regex = try! NSRegularExpression(pattern: "[0-9a-f]{1,2}", options: .caseInsensitive)
```

Comment:
Let's avoid unwrapping - wrap it in the `if let's` or `guard let's`  and when condition is not meet - return nil. This we'll be a lot safer.